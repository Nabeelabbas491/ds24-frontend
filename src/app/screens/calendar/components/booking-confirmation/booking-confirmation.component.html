<div
  class="bg-neutral-n0 w-100 my-0 mx-auto flex flex-col justify-start items-center py-5 px-10"
  *ngIf="bookingSuccess$ | async as bookingSuccess"
>
  <ds-icon
    [name]="icons.Checkmark"
    [color]="iconColors.Success300"
    [size]="48"
    [title]="'CALENDAR.PREVIOUS_MONTH' | translate"
  ></ds-icon>
  <h3 class="text-neutral-n500 font-medium mt-5 break-words w-full text-center">
    {{ 'BOOKING.BOOKING_CONFIRMATION.BOOKING_NAME' | translate: { bookingName: bookingSuccess.name } }},
    {{ 'BOOKING.BOOKING_CONFIRMATION.BOOKING_CONFIRMED' | translate }}
  </h3>
  <h4 class="font-medium mt-5">
    {{ bookingSuccess.dayTitle | date: 'fullDate' }} - {{ bookingSuccess.timeSlot | timeSlot }}
  </h4>
  <p class="text-neutral-n500 mt-5">{{ 'BOOKING.BOOKING_CONFIRMATION.CALENDAR_INVITATION_SENT' | translate }}</p>
  <a class="cursor-pointer underline mt-3 text-base" href="#">{{
    'BOOKING.BOOKING_CONFIRMATION.SEND_CONFIRMATION_EMAIL_AGAIN' | translate
  }}</a>
  <div class="my-5" *ngIf="bookingSuccess.meetingTypeDetail?.id === meetingTypeIds.Zoom">
    <h4 class="font-medium my-3 text-center">{{ 'BOOKING.BOOKING_CONFIRMATION.LINK_TO_ZOOM_MEETING' | translate }}</h4>
    <ds-copy-info-box
      [copyText]="bookingSuccess.zoomLink"
      (click)="copyZoomLink(bookingSuccess.zoomLink)"
    ></ds-copy-info-box>
  </div>
  <div class="my-5" *ngIf="bookingSuccess.meetingTypeDetail?.id === meetingTypeIds.Inbound">
    <h4 class="font-medium my-3 text-center">{{ 'BOOKING.BOOKING_CONFIRMATION.VENDOR_PHONE_NUMBER' | translate }}</h4>
    <ds-copy-info-box
      [copyText]="bookingSuccess.vendorPhoneNumber"
      (click)="copyPhone(bookingSuccess.vendorPhoneNumber)"
    ></ds-copy-info-box>
  </div>
  <div class="mt-3 flex justify-start flex-row mobile-view:flex-col">
    <button dsButton="default" class="mr-2 mobile-view:w-full" (click)="downloadICS(bookingSuccess.appointmentId)">
      {{ 'BOOKING.BOOKING_CONFIRMATION.DONWNLOAD_ICS' | translate }}
    </button>
    <button
      dsButton="default"
      class="mobile-view:w-full mobile-view:mt-3"
      [disabled]="(isBookingSynced$ | async) === true"
      (click)="saveToGoogleCalendar()"
    >
      {{ 'BOOKING.BOOKING_CONFIRMATION.SAVE_TO_GOOGLE_CALENDAR' | translate }}
    </button>
  </div>
  <div class="flex flex-row justify-center mt-3">
    <ds-spinner
      [color]="spinnerColor"
      class="mx-auto"
      *ngIf="(downloadClientICSPending$ | async) || (getAuthenticationUrlPending$ | async)"
      [size]="30"
    ></ds-spinner>
  </div>
</div>
