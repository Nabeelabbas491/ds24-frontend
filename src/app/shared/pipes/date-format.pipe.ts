import { Pipe, PipeTransform } from '@angular/core';
import { DateFormat } from './../../types/misc.type';
import { TranslateService } from '@ngx-translate/core';
import { format, parseISO } from 'date-fns';

@Pipe({
  name: 'dateFormat',
})
export class DateFormatPipe implements PipeTransform {
  constructor(private translateService: TranslateService) {}
  transform(dateString: string | undefined | null, formatStr: DateFormat = 'full'): string {
    if (dateString == null) {
      return '';
    }
    if (formatStr === 'full') {
      return format(parseISO(dateString), this.translateService.instant('CALENDAR.DAY_FORMAT_STRING.FULL'));
    }
    if (formatStr === 'full_without_year') {
      return format(
        parseISO(dateString),
        this.translateService.instant('CALENDAR.DAY_FORMAT_STRING.FULL_WITHOUT_YEAR'),
      );
    }

    if (formatStr === 'day_month_year') {
      return format(parseISO(dateString), 'dd MMMM, yyyy');
    }
    if (formatStr === 'day_half-month_year') {
      return format(parseISO(dateString), 'dd MMM, yyyy');
    }

    return '';
  }
}
