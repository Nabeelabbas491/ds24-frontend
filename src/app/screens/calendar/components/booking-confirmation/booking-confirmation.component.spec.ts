import { ComponentFixture, TestBed } from '@angular/core/testing';

import { BookingConfirmationComponent } from './booking-confirmation.component';
import { TranslateFakeLoader, TranslateLoader, TranslateModule, TranslateService } from '@ngx-translate/core';
import { IconModule } from '@ds24/elements';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { PipesModule } from '../../../../shared/pipes/pipes.module';
import { State } from '../../../../store/state';
import { of } from 'rxjs';
import { TranslateMockPipe } from './../../../../shared/pipes/translate-mock.pipe';
import { actions } from './../../../../store';
import { Clipboard } from '@angular/cdk/clipboard';

describe('BookingConfirmationComponent', () => {
  let component: BookingConfirmationComponent;
  let fixture: ComponentFixture<BookingConfirmationComponent>;
  let mockStore: MockStore<State>;

  const clipboardMock = {
    copy: jest.fn(),
  };

  const mockTranslateService = {
    get: jest.fn().mockImplementation(() => of('sample message')),
    instant: jest.fn().mockReturnValue('sample message'),
  };

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [BookingConfirmationComponent, TranslateMockPipe],
      providers: [
        { provide: Clipboard, useValue: clipboardMock },
        provideMockStore(),
        { provide: TranslateService, useValue: mockTranslateService },
      ],
      imports: [
        TranslateModule.forRoot({
          loader: {
            provide: TranslateLoader,
            useClass: TranslateFakeLoader,
          },
        }),
        IconModule,
        PipesModule,
      ],
    });
    fixture = TestBed.createComponent(BookingConfirmationComponent);
    component = fixture.componentInstance;
    mockStore = TestBed.inject(MockStore);
    jest.spyOn(mockStore, 'dispatch');
    fixture.detectChanges();
  });

  it('should compile', () => {
    fixture.detectChanges();
    expect(component).toMatchSnapshot();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should copy vendor phone number to clipboard', () => {
    component.copyPhone('+1234567');
    expect(clipboardMock.copy).toHaveBeenCalledWith('+1234567');
  });

  it('should copy Zoom link to clipboard', () => {
    component.copyZoomLink('http://test.com');
    expect(clipboardMock.copy).toHaveBeenCalledWith('http://test.com');
  });

  it('should call the snackbar popup', () => {
    component.showSnackbarMessage('SAMPLE_TRANSLATION');
    fixture.detectChanges();
    expect(mockStore.dispatch).toHaveBeenCalledWith(
      actions.notification.callSnackBarPopup({ snackBarMessage: 'sample message' }),
    );
  });

  it('should call the translate service', () => {
    const translation = 'BOOKING.BOOKING_CONFIRMATION.VENDOR_PHONE_NUMBER_COPIED';
    component.showSnackbarMessage(translation);
    fixture.detectChanges();
    expect(mockTranslateService.instant).toHaveBeenCalledWith(translation);
  });
});
