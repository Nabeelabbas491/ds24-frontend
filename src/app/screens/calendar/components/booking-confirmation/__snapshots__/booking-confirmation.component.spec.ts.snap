// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingConfirmationComponent should compile 1`] = `
BookingConfirmationComponent {
  "__ngContext__": 0,
  "_document": Document {
    "location": Location {
      "assign": [Function],
      "hash": "",
      "host": "localhost",
      "hostname": "localhost",
      "href": "http://localhost/",
      "origin": "http://localhost",
      "pathname": "/",
      "port": "",
      "protocol": "http:",
      "reload": [Function],
      "replace": [Function],
      "search": "",
      "toString": [Function],
    },
  },
  "_translate": {
    "get": [MockFunction],
    "instant": [MockFunction],
  },
  "bookingProductId$": _Store {
    "actionsObserver": _ActionsSubject {
      "_value": {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "operator": [Function],
    "reducerManager": _MockReducerManager {
      "_value": [Function],
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "source": _Store {
      "actionsObserver": _ActionsSubject {
        "_value": {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "operator": [Function],
      "reducerManager": _MockReducerManager {
        "_value": [Function],
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "source": _MockStore {
        "actionsObserver": _ActionsSubject {
          "_value": {
            "type": "@ngrx/store/init",
          },
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "dispatch": [MockFunction],
        "initialState": {},
        "lastState": {},
        "reducerManager": _MockReducerManager {
          "_value": [Function],
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "scannedActions$": Observable {
          "source": _ActionsSubject {
            "_value": {
              "type": "@ngrx/store/init",
            },
            "closed": false,
            "currentObservers": null,
            "hasError": false,
            "isStopped": false,
            "observers": [],
            "thrownError": null,
          },
        },
        "selectors": Map {},
        "source": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
        "state": [Function],
        "state$": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
      },
      "state": [Function],
    },
    "state": [Function],
  },
  "bookingSnapshot$": _Store {
    "actionsObserver": _ActionsSubject {
      "_value": {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "operator": [Function],
    "reducerManager": _MockReducerManager {
      "_value": [Function],
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "source": _Store {
      "actionsObserver": _ActionsSubject {
        "_value": {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "operator": [Function],
      "reducerManager": _MockReducerManager {
        "_value": [Function],
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "source": _MockStore {
        "actionsObserver": _ActionsSubject {
          "_value": {
            "type": "@ngrx/store/init",
          },
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "dispatch": [MockFunction],
        "initialState": {},
        "lastState": {},
        "reducerManager": _MockReducerManager {
          "_value": [Function],
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "scannedActions$": Observable {
          "source": _ActionsSubject {
            "_value": {
              "type": "@ngrx/store/init",
            },
            "closed": false,
            "currentObservers": null,
            "hasError": false,
            "isStopped": false,
            "observers": [],
            "thrownError": null,
          },
        },
        "selectors": Map {},
        "source": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
        "state": [Function],
        "state$": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
      },
      "state": [Function],
    },
    "state": [Function],
  },
  "bookingSuccess$": _Store {
    "actionsObserver": _ActionsSubject {
      "_value": {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "operator": [Function],
    "reducerManager": _MockReducerManager {
      "_value": [Function],
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "source": _Store {
      "actionsObserver": _ActionsSubject {
        "_value": {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "operator": [Function],
      "reducerManager": _MockReducerManager {
        "_value": [Function],
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "source": _MockStore {
        "actionsObserver": _ActionsSubject {
          "_value": {
            "type": "@ngrx/store/init",
          },
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "dispatch": [MockFunction],
        "initialState": {},
        "lastState": {},
        "reducerManager": _MockReducerManager {
          "_value": [Function],
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "scannedActions$": Observable {
          "source": _ActionsSubject {
            "_value": {
              "type": "@ngrx/store/init",
            },
            "closed": false,
            "currentObservers": null,
            "hasError": false,
            "isStopped": false,
            "observers": [],
            "thrownError": null,
          },
        },
        "selectors": Map {},
        "source": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
        "state": [Function],
        "state$": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
      },
      "state": [Function],
    },
    "state": [Function],
  },
  "clipboard": {
    "copy": [MockFunction],
  },
  "downloadClientICSPending$": _Store {
    "actionsObserver": _ActionsSubject {
      "_value": {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "operator": [Function],
    "reducerManager": _MockReducerManager {
      "_value": [Function],
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "source": _Store {
      "actionsObserver": _ActionsSubject {
        "_value": {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "operator": [Function],
      "reducerManager": _MockReducerManager {
        "_value": [Function],
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "source": _MockStore {
        "actionsObserver": _ActionsSubject {
          "_value": {
            "type": "@ngrx/store/init",
          },
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "dispatch": [MockFunction],
        "initialState": {},
        "lastState": {},
        "reducerManager": _MockReducerManager {
          "_value": [Function],
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "scannedActions$": Observable {
          "source": _ActionsSubject {
            "_value": {
              "type": "@ngrx/store/init",
            },
            "closed": false,
            "currentObservers": null,
            "hasError": false,
            "isStopped": false,
            "observers": [],
            "thrownError": null,
          },
        },
        "selectors": Map {},
        "source": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
        "state": [Function],
        "state$": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
      },
      "state": [Function],
    },
    "state": [Function],
  },
  "getAuthenticationUrlPending$": _Store {
    "actionsObserver": _ActionsSubject {
      "_value": {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "operator": [Function],
    "reducerManager": _MockReducerManager {
      "_value": [Function],
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "source": _Store {
      "actionsObserver": _ActionsSubject {
        "_value": {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "operator": [Function],
      "reducerManager": _MockReducerManager {
        "_value": [Function],
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "source": _MockStore {
        "actionsObserver": _ActionsSubject {
          "_value": {
            "type": "@ngrx/store/init",
          },
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "dispatch": [MockFunction],
        "initialState": {},
        "lastState": {},
        "reducerManager": _MockReducerManager {
          "_value": [Function],
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "scannedActions$": Observable {
          "source": _ActionsSubject {
            "_value": {
              "type": "@ngrx/store/init",
            },
            "closed": false,
            "currentObservers": null,
            "hasError": false,
            "isStopped": false,
            "observers": [],
            "thrownError": null,
          },
        },
        "selectors": Map {},
        "source": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
        "state": [Function],
        "state$": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
      },
      "state": [Function],
    },
    "state": [Function],
  },
  "iconColors": {
    "Danger300": "fill-danger-e300",
    "Neutral0": "fill-neutral-n0",
    "Neutral200": "fill-neutral-n200",
    "Neutral300": "fill-neutral-n300",
    "Neutral400": "fill-neutral-n400",
    "Neutral500": "fill-neutral-n500",
    "Primary100": "fill-primary-p100",
    "Primary300": "fill-primary-p300-main",
    "Primary400": "fill-primary-p400",
    "Success300": "fill-success-s300",
    "Warning300": "fill-warning-w300",
    "white": "fill-white",
  },
  "icons": {
    "ArrowDown": "arrow-down",
    "ArrowLeft": "arrow-left",
    "ArrowUp": "arrow-up",
    "Calendar": "calendar",
    "Camera": "camera",
    "CaretDown": "caret-down",
    "Checked": "checked",
    "Checkmark": "checkmark",
    "Chevron": "chevron",
    "Clear": "clear",
    "Clock": "clock",
    "CoinHands": "coin-hands",
    "Columns": "columns",
    "Copy": "copy",
    "Desktop": "desktop",
    "DialogClose": "dialog-close",
    "DigicallsOrderSales": "digicalls-order-sales",
    "DragAndDrop": "drag-and-drop",
    "ErrorFilled": "error-filled",
    "ExclamationMark": "exclamation-mark",
    "ExclamationMarkFilled": "exclamation-mark-filled",
    "ExternalLink": "external-link",
    "Eye": "eye",
    "EyeCrossed": "eye-crossed",
    "Faq": "faq",
    "File": "file",
    "FilePlay": "file-play",
    "Folder": "folder",
    "Funnel": "funnel",
    "Gear": "gear",
    "Hamburger": "hamburger",
    "InfoFilled": "info-filled",
    "Link": "link",
    "LinkHorizontal": "link-horizontal",
    "Loader": "loader",
    "MagnifyingGlass": "magnifying-glass",
    "Members": "members",
    "Minus": "minus",
    "Mobile": "mobile",
    "More": "more",
    "Pencil": "pencil",
    "Phone": "phone",
    "Picture": "picture",
    "Plug": "plug",
    "Plus": "plus",
    "Popup": "popup",
    "QuestionMark": "question-mark",
    "Redo": "redo",
    "Restore": "restore",
    "Rows": "rows",
    "Save": "save",
    "Sections": "sections",
    "Skip": "skip",
    "Stack": "stack",
    "Stats": "stats",
    "Tablet": "tablet",
    "Tag": "tag",
    "Text": "text",
    "Translation": "translation",
    "Trash": "trash",
    "Undo": "undo",
    "UserMoney": "user-money",
  },
  "isBookingSynced$": _Store {
    "actionsObserver": _ActionsSubject {
      "_value": {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "operator": [Function],
    "reducerManager": _MockReducerManager {
      "_value": [Function],
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "source": _Store {
      "actionsObserver": _ActionsSubject {
        "_value": {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "operator": [Function],
      "reducerManager": _MockReducerManager {
        "_value": [Function],
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
      "source": _MockStore {
        "actionsObserver": _ActionsSubject {
          "_value": {
            "type": "@ngrx/store/init",
          },
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "dispatch": [MockFunction],
        "initialState": {},
        "lastState": {},
        "reducerManager": _MockReducerManager {
          "_value": [Function],
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [],
          "thrownError": null,
        },
        "scannedActions$": Observable {
          "source": _ActionsSubject {
            "_value": {
              "type": "@ngrx/store/init",
            },
            "closed": false,
            "currentObservers": null,
            "hasError": false,
            "isStopped": false,
            "observers": [],
            "thrownError": null,
          },
        },
        "selectors": Map {},
        "source": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
        "state": [Function],
        "state$": _MockState {
          "_value": {},
          "closed": false,
          "currentObservers": null,
          "hasError": false,
          "isStopped": false,
          "observers": [
            SafeSubscriber {
              "_finalizers": [
                Subscription {
                  "_finalizers": null,
                  "_parentage": [Circular],
                  "closed": false,
                  "initialTeardown": [Function],
                },
              ],
              "_parentage": null,
              "closed": false,
              "destination": ConsumerObserver {
                "partialObserver": {
                  "error": [Function],
                  "next": [Function],
                },
              },
              "initialTeardown": undefined,
              "isStopped": false,
            },
          ],
          "state": [Function],
          "thrownError": null,
        },
      },
      "state": [Function],
    },
    "state": [Function],
  },
  "meetingTypeIds": {
    "1": "Inbound",
    "2": "Outbound",
    "3": "Zoom",
    "Inbound": 1,
    "Outbound": 2,
    "Zoom": 3,
  },
  "ngAfterContentChecked$": Subject {
    "closed": false,
    "currentObservers": [],
    "hasError": false,
    "isStopped": false,
    "observers": [],
    "thrownError": null,
  },
  "ngAfterContentInit$": Subject {
    "closed": false,
    "currentObservers": [],
    "hasError": false,
    "isStopped": true,
    "observers": [],
    "thrownError": null,
  },
  "ngAfterViewChecked$": Subject {
    "closed": false,
    "currentObservers": [],
    "hasError": false,
    "isStopped": false,
    "observers": [],
    "thrownError": null,
  },
  "ngAfterViewInit$": Subject {
    "closed": false,
    "currentObservers": [],
    "hasError": false,
    "isStopped": true,
    "observers": [],
    "thrownError": null,
  },
  "ngDoCheck$": Subject {
    "closed": false,
    "currentObservers": [],
    "hasError": false,
    "isStopped": false,
    "observers": [],
    "thrownError": null,
  },
  "ngOnChanges$": Subject {
    "closed": false,
    "currentObservers": null,
    "hasError": false,
    "isStopped": false,
    "observers": [],
    "thrownError": null,
  },
  "ngOnDestroy$": Subject {
    "closed": false,
    "currentObservers": null,
    "hasError": false,
    "isStopped": false,
    "observers": [],
    "thrownError": null,
  },
  "ngOnInit$": Subject {
    "closed": false,
    "currentObservers": [],
    "hasError": false,
    "isStopped": true,
    "observers": [],
    "thrownError": null,
  },
  "spinnerColor": "fill-primary-p300-main",
  "store": _MockStore {
    "actionsObserver": _ActionsSubject {
      "_value": {
        "type": "@ngrx/store/init",
      },
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "dispatch": [MockFunction],
    "initialState": {},
    "lastState": {},
    "reducerManager": _MockReducerManager {
      "_value": [Function],
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [],
      "thrownError": null,
    },
    "scannedActions$": Observable {
      "source": _ActionsSubject {
        "_value": {
          "type": "@ngrx/store/init",
        },
        "closed": false,
        "currentObservers": null,
        "hasError": false,
        "isStopped": false,
        "observers": [],
        "thrownError": null,
      },
    },
    "selectors": Map {},
    "source": _MockState {
      "_value": {},
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [
        SafeSubscriber {
          "_finalizers": [
            Subscription {
              "_finalizers": null,
              "_parentage": [Circular],
              "closed": false,
              "initialTeardown": [Function],
            },
          ],
          "_parentage": null,
          "closed": false,
          "destination": ConsumerObserver {
            "partialObserver": {
              "error": [Function],
              "next": [Function],
            },
          },
          "initialTeardown": undefined,
          "isStopped": false,
        },
      ],
      "state": [Function],
      "thrownError": null,
    },
    "state": [Function],
    "state$": _MockState {
      "_value": {},
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": [
        SafeSubscriber {
          "_finalizers": [
            Subscription {
              "_finalizers": null,
              "_parentage": [Circular],
              "closed": false,
              "initialTeardown": [Function],
            },
          ],
          "_parentage": null,
          "closed": false,
          "destination": ConsumerObserver {
            "partialObserver": {
              "error": [Function],
              "next": [Function],
            },
          },
          "initialTeardown": undefined,
          "isStopped": false,
        },
      ],
      "state": [Function],
      "thrownError": null,
    },
  },
}
`;
